File: src/AutoRest.SdkExplorer/Model/Schema/SchemaNone.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Text.Json.Serialization;

namespace AutoRest.SdkExplorer.Model.Schema
{
    public class SchemaNone : SchemaBase
    {
        public const string SCHEMA_TYPE = "NONE_SCHEMA";

        public string Reason { get; set; } = string.Empty;

        private SchemaNone()
            : base("OnlyForYamlDeserializor", SCHEMA_TYPE)
        {
        }

        [JsonConstructor]
        public SchemaNone(string schemaKey)
            : base(schemaKey, SCHEMA_TYPE)
        {
        }
    }
}

File: src/AutoRest.SdkExplorer.TypeScript/src/Model/Schema/SchemaNone.ts
import { SchemaBase } from "./SchemaBase";

export class SchemaNone extends SchemaBase implements AutoRest.SdkExplorer.Interface.SchemaNone {
    reason: string;

    constructor(data: AutoRest.SdkExplorer.Interface.SchemaNone) {
        super(data);
        this.reason = data.reason!;
    }
}
